src/entities/removeLiquidity/removeLiquidityV3/encodeRemoveLiquiditySingleTokenExactOut.ts

```diff
@@ -1,16 +1,20 @@
-import { removeLiquiditySingleTokenExactInShouldHaveTokenOutIndexError } from '@/utils';
-import { RemoveLiquidityBaseBuildCallInput } from '../types';
 import { encodeFunctionData } from 'viem';
 import { balancerRouterAbiExtended } from '@/abi';
 import { Hex } from '@/types';
+import { missingParameterError } from '@/utils';
+import { RemoveLiquidityBaseBuildCallInput } from '../types';
 
 export const encodeRemoveLiquiditySingleTokenExactOut = (
     input: RemoveLiquidityBaseBuildCallInput & { userData: Hex },
```

Comment:
Can we use the new `RemoveLiquidityV3BuildCallInput` type here as well?

Reply:
@brunoguerios if you do a global search across files it looks like there are a few more spots that could be tidied up

```
RemoveLiquidityBaseBuildCallInput & { userData: Hex }
```

Reply:
Thanks for the heads up! 
This change was the one I had in mind when I said I needed to take care or the PR would grow out of scope 😬 
I'll move on with the changes since I don't think it's worth going back now, but this will change again once we make userData optional (as I suggested on slack).

Reply:
oh my bad I interpreted the type refactor as a "[small cleanup](https://www.notion.so/SDK-Guidelines-1ace395e22818093b1eafa1813ca5e5b)" but maybe there's something Im not understanding

Reply:
I consider moving `userData` to inside the new `RemoveLiquidityV3BuildCallInput` type a simple cleanup 👍 - but it does add unnecessary complexity to an already huge PR.
I realized that when I was halfway through the changes and decided to keep going, but I'll try to avoid those in my next PRs.
Ps: I don't consider making `userData` optional a small cleanup, as it changes behavior, so I'll save it for a following PR